kafka-client/src/main/java/dev/responsive/kafka/internal/clients/ResponsiveGlobalConsumer.java

```diff
@@ -182,7 +183,8 @@ static SingletonConsumerRecords of(final ConsumerRecords<byte[], byte[]> records
     public SingletonConsumerRecords(
         final Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> records
     ) {
-      super(records);
+      super(records, Collections.emptyMap());
+      // TODO(sophie): need to pass in the actual next offsets here?
```

Comment:
Related to the upgrade -- a recent KIP added offset tracking to the consumer records, haven't looked into it much yet but I suspect Streams may be using this in some way and so we might need to actually implement this